src/Reflection/ReflectionFunctionAbstract.php

```diff
@@ -396,7 +396,9 @@ private function createReturnType(): ReflectionNamedType|ReflectionUnionType|Ref
             return null;
         }
 
-        return ReflectionType::createFromNode($returnType);
+        assert($this instanceof ReflectionMethod || $this instanceof ReflectionFunction);
```

Comment:
`$this` is a `ReflectionFunctionAbstract`, which already has `$this->reflector`

Why is an `assert()` needed?

Reply:
I'm trying to use `ReflectionMethod|ReflectionFunction` as parameter type hint when possible and Psalm/PHPStan are not happy with `ReflectionFunctionAbstract` so I need to tell them it's `ReflectionMethod` or `ReflectionFunction`.

The best solution would be to get rid of `ReflectionFunctionAbstract` completly.

Reply:
Ah, makes sense, thanks! :) 


 > The best solution would be to get rid of `ReflectionFunctionAbstract` completly.

Possibly long-term - IMO irrelevant for the `5.0.0` cut-off

Reply:
We'll see how fast I can prepar PR :D

Reply:
Please don't overload yourself, especially with refactoring exercises that use your wrists more than your mind: we can gladly defer more BC breaks to a `6.0.0`, as there's nothing preventing that